include/imterm/terminal.hpp

```diff
@@ -80,6 +80,10 @@ namespace ImTerm {
 		using buffer_type = std::array<char, 1024>;
 		using small_buffer_type = std::array<char, 128>;
 
+		std::atomic_flag m_flag;
+#define IMTERM_LOCK() while (m_flag.test_and_set(std::memory_order_seq_cst)) {}
+#define IMTERM_UNLOCK() m_flag.clear(std::memory_order_seq_cst)
```

Comment:
I’d rather see a method instead of a macro, but if you mind using a method, then please `#undef IMTERM_LOCK` and `IMTERM_UNLOCK` at the end of the file